src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -664,6 +664,13 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
 
     auto fmt_map = get_preferred_formats(p, lo);
 
+    for (auto n : p.get_processing_order()) {
+        auto forced_format = lo.get_forced_format(*n);
+        if (forced_format != format::any) {
+            n->set_preferred_output_fmt(0, forced_format);
+        }
+    }
```

Comment:
If it is targeting onednn, it is better implemented in `select_preferred_format` pass

Reply:
Done.